### PR TITLE
Prevent developers from having several active api keys at the same time

### LINKED ISSUE #8211: Some developers have more than one active API key
Some developers have more than one active API key, which is not supported and throws errors since the retrieval function relies on having only one.

We should enforce that in the database if possible, and figure out what to do with the affected keys.

### PR DESCRIPTION
**Pull request by diox**
Fix #8211

### CODE REVIEW

File: src/olympia/migrations/1016-prevent-duplicate-api-keys.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+-- First, disable all active API keys from users who have multiple active API
+-- keys. We can't know which one they expected to work, and they couldn't use
+-- them before anyway because of MultipleObjectsReturned exception.
+-- The double subquery is here to let MySQL update and select from the same
+-- table by creating a temporary one.
+UPDATE `api_key` SET `is_active` = 0 WHERE `user_id` IN (
+    SELECT `user_id` FROM (
+        SELECT `user_id` FROM `api_key`
+        GROUP BY `user_id` HAVING SUM(is_active) > 1
+        ORDER BY `user_id`)
```
Comment (wagnerand) [inline src/olympia/migrations/1016-prevent-duplicate-api-keys.sql]: Nit: Removing the `ORDER BY` clause saves us a few ms.

File: src/olympia/api/models.py [hunk 2]
```diff
@@ -23,7 +23,7 @@ class APIKey(ModelBase):
     A developer's key/secret pair to access the API.
     """
     user = models.ForeignKey(UserProfile, related_name='api_keys')
-    is_active = models.BooleanField(default=True)
+    is_active = models.NullBooleanField(default=True)
```
Comment (eviljeff) [inline src/olympia/api/models.py]: I had to read through the whole patch to realise _why_ `BooleanField` was changed to `NullBooleanField` (the constraint) - can you add a comment here about it?

Comment (diox) [reply src/olympia/api/models.py]: Sure, done in dcfe291086ce72fc9e5a1affe92a24453223c5e1

File: src/olympia/migrations/1016-prevent-duplicate-api-keys.sql (hunk 1 above)
Comment (diox) [reply src/olympia/migrations/1016-prevent-duplicate-api-keys.sql]: probably not even ms :)

Comment (wagnerand) [reply src/olympia/migrations/1016-prevent-duplicate-api-keys.sql]: Actually, more than just a few:

I just tried on prod and it took ~500ms including ordering and 200ms without.

**Pull request merged**